Added support for Rockhip FriendlyElec NanoPC T6 LTS

gonciarz opened the pull request:
FriendlyElec released two versions of their NanoPC T6 board non-LTS (2301) and LTS (2310).
This PR adds support for LTS board.
Non-LTS image works fine with LTS board. This PR just adapt DTS files and provide official support.
The original patch: https://lore.kernel.org/all/20240829-friendlyelec-nanopc-t6-lts-v6-0-edff247e8c02@linaro.org/
was modified and adjusted to mainstream kernel 6.12. The previous version included:
`#include "rk3588-nanopc-t6.dts"`
and a modified one:
`#include "rk3588-nanopc-t6.dtsi`

gonciarz commented:
PR still requires few changes. uboot compilation fails, because it expects nanopc-t6-lts-rk3588_defconfig.
What would be the most convenient way to create it?
```
Applying /sync/workspace/openwrt3/openwrt/package/boot/uboot-rockchip/patches/105-nanopc-t4-fix-memory-unstability.patch using plaintext: 
patching file arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi
touch /sync/workspace/openwrt3/openwrt/build_dir/target-aarch64_generic_musl/u-boot-nanopc-t6-lts-rk3588/u-boot-2024.10/.prepared_ea7339b0cae1065ba767a16b4cbc20dd_6664517399ebbbc92a37c5bb081b5c53
rm -f /sync/workspace/openwrt3/openwrt/build_dir/target-aarch64_generic_musl/u-boot-nanopc-t6-lts-rk3588/u-boot-2024.10/.configured_*
rm -f /sync/workspace/openwrt3/openwrt/staging_dir/target-aarch64_generic_musl/stamp/.uboot-rockchip.nanopc-t6-lts-rk3588_installed
make  -C /sync/workspace/openwrt3/openwrt/build_dir/target-aarch64_generic_musl/u-boot-nanopc-t6-lts-rk3588/u-boot-2024.10 CROSS_COMPILE=aarch64-openwrt-linux-musl- USE_PRIVATE_LIBGCC=yes nanopc-t6-lts-rk3588_config
make[4]: Entering directory '/sync/workspace/openwrt3/openwrt/build_dir/target-aarch64_generic_musl/u-boot-nanopc-t6-lts-rk3588/u-boot-2024.10'
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  YACC    scripts/kconfig/zconf.tab.c
  LEX     scripts/kconfig/zconf.lex.c
  HOSTCC  scripts/kconfig/zconf.tab.o
  HOSTLD  scripts/kconfig/conf
cc1: fatal error: ./arch/../configs/nanopc-t6-lts-rk3588_defconfig: No such file or directory
```

1715173329 commented:
You should backport upstream commits instead:
For u-boot: https://github.com/u-boot/u-boot/commit/7cec3e701940064b2cfc0cf8b80ff24c391c55ec
For kernel: https://github.com/torvalds/linux/commit/db1dcbe5f752d423421f77d54d246398b196f670

hauke commented:
Please refresh the patches, see:
```
Run . .github/workflows/scripts/ci_helpers.sh
target/linux/rockchip/patches-6.6/129-v6.12-arm64-dts-rockchip-friendlyelec-nanopc-t6-lts-board.patch
Kernel patches for rockchip/armv8 require refresh. (run 'make target/linux/refresh' and force push this pr)
You can also check the provided artifacts with the refreshed patch from this CI run.
Error: Process completed with exit code 1.
```

Run:
```
make target/{clean,refresh} V=99
make package/uboot-rockchip/{clean,refresh} V=99
```

hauke commented:
Please add the patch header too.
Your can extract it from github by just adding a .patch to the end of the URL:
https://github.com/torvalds/linux/commit/db1dcbe5f752d423421f77d54d246398b196f670.patch
https://github.com/u-boot/u-boot/commit/7cec3e701940064b2cfc0cf8b80ff24c391c55ec.patch

gonciarz committed:
rockchip: Add support for Rockchip NanoPC T6 LTS

CPU: Rockchip RK3588, Quad-core Cortex-A76 (up to 2.4GHz) and Quad-core Cortex-A55 CPU (up to 1.8GHz)
RAM: 4GB/8GB/16GB LPDDR4X (@2133MHz)
Ethernet: 2 x PCIe 2.5G Ethernet
Storage: eMMC Flash: 32GB/64GB/256GB eMMC (at HS400 mode)
Storage: Nor Flash: 32MB SPI Nor Flash
Storage: microSD: support up to SDR104 mode
Storage: M.2: 1x M-Key PCIe 3.0x4 (for NVMe SSDs up to 2,500 MB/s), 1x E-key PCIe 2.1x1 and USB2.0 Host
USB: 1x USB 3.0 Type-A, 1x USB 3.0 Type-C (with DP 4Kp60), 2x USB 2.0 Type-A
Debug: USB-C to Debug UART
2x GPIO Controlled LED (SYS, LED1)

Others:
GPU: Mali-G610 MP4, compatible with OpenGLES 1.1, 2.0, and 3.2, OpenCL up to 2.2 and Vulkan1.2
VPU: 8K@60fps H.265 and VP9 decoder, 8K@30fps H.264 decoder, 4K@60fps AV1 decoder, 8K@30fps H.264 and H.265 encoder
NPU: 6TOPs, supports INT4/INT8/INT16/FP16
Video input: 1x HDMI 2.0 (4Kp60), 2x 4-lane MIPI-CSI (compatible with MIPI V1.2)
Video output: 1x HDMI 2.1 (8Kp60), 1x HDMI 2.0 (4Kp60), 2x 4-lane MIPI-DSI (compatible with MIPI DPHY 2.0 or CPHY 1.1)
Audio: 1x 3.5mm jack (stereo output), 1x 2.0mm PH-2A connector for analog microphone input
GPIO: 40-pin 2.54mm header connector (up to 2x SPIs, 6x UARTs, 1x I2Cs, 8x PWMs, 2x I2Ss, 28x GPIOs)
10-Pin 2.54mm header connector including Debug UART(3.3V TTL) and 2x USB 2.0 Host
2-Pin 1.27/1.25mm RTC battery input connector for low power RTC IC HYM8563TS
one 38Khz IR receiver
MASK button for eMMC update, reset button, and Power button one 5V Fan connector
Power supply: 5.5*2.1mm DC Jack & 2-Pin 3.5mm pitch connector, 12VDC input.

Signed-off-by: Robert Gonciarz <gonciarz@gmail.com>

gonciarz committed:
Merge branch 'main' into rockhip-friendlyelec-nanopc-t6-lts
